lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -5925,6 +5927,12 @@ void TypeChecker::checkConformancesInContext(DeclContext *dc,
                    emitWarning ? diag::nscoding_unstable_mangled_name_warn
                                : diag::nscoding_unstable_mangled_name,
                    *kind, classDecl->TypeDecl::getDeclaredInterfaceType());
+          if (isFixable) {
+            diagnose(classDecl, diag::add_nskeyedarchivelegacy_attr)
+              .fixItInsert(
+                classDecl->getAttributeInsertionLoc(/*forModifier=*/false),
+                "@NSKeyedArchiveLegacy(\"<#class archival name#>\")");
```

Comment:
I don't think so, but if you only care about non-generic classes I would extract the right bits from the old mangler. It should not amount to much.